framework/mongo/utils.py

```diff
@@ -57,3 +59,11 @@ def wrapper(cls):
         ])
         return cls
     return wrapper
+
+
+def get_or_http_not_found(Model, pk):
```

Comment:
If I remember correctly, you previously handled deleted nodes by returning a HTTP Gone response. Was this purposefully removed? I thought it was useful.

How about

```python
def get_or_http_not_found(Model, pk):
    instance = Model.load(pk)
    if getattr(instance, 'is_deleted', False):
        raise HTTPError(http.GONE)
    if not instance:
        raise HTTPError(http.NOT_FOUND)
    else:
        return instance
```